src/main/java/org/tugraz/sysds/runtime/controlprogram/context/SparkExecutionContext.java

```diff
@@ -1132,26 +1123,21 @@ public static MatrixBlock toMatrixBlock(JavaPairRDD<MatrixIndexes, MatrixCell> r
 	}
 
 	public static TensorBlock toTensorBlock(JavaPairRDD<TensorIndexes, TensorBlock> rdd, DataCharacteristics dc) {
-		// TODO support DataTensors
 		long t0 = DMLScript.STATISTICS ? System.nanoTime() : 0;
 
 		// TODO special case single block
-		int[] idims = new int[dc.getNumDims()];
-		for (int i = 0; i < idims.length; i++) {
-			idims[i] = (int)dc.getDim(i);
-		}
+		int[] idims = Arrays.stream(dc.getDims()).mapToInt(i -> (int) i).toArray();
```

Comment:
ok, then maybe getLongDims and getIntDims where getDims simply calls getLongDims?